resources.go

```diff
@@ -921,7 +921,12 @@ func Provider() tfbridge.ProviderInfo {
 				// deletes the same attachment we just created, since it is structurally equivalent!
 				DeleteBeforeReplace: true,
 			},
-			"aws_iam_role_policy": {Tok: awsResource(iamMod, "RolePolicy")},
+			"aws_iam_role_policy": {
+				Tok: awsResource(iamMod, "RolePolicy"),
+				Fields: map[string]*tfbridge.SchemaInfo{
+					"role": {Type: awsType(iamMod+"/role", "Role")},
```

Comment:
I was in fact just talking to @pgavlin about this sort of thing yesterday, as the places where we have strong types can trip up the HCL converter. I believe anywhere we add schema types that aren't just string typedefs, we should do what Matt is suggesting, so we can still accept strings.

Reply:
This has the nice consequence that the dozens (hundreds?) of places we've not yet added the strong typing can be patched up incrementally over time without any breaking changes.

Reply:
Yes - we should use `AltTypes` here to avoid the breaking change.  We should ideally do the same in the existing places where `Role` is used as a strongly typed override.

An example of the "right" pattern is here: https://github.com/pulumi/pulumi-aws/blob/44eed4bfe4a916e7fc71b9d56c2ae73f8a7f9127/resources.go#L437-L442.

This is a good reminder to write up the API guidelines for these projections - we made a set of decisions about how a few related topics a few months back but doesn't appears we never fully documented that.  Opened https://github.com/pulumi/pulumi-terraform/issues/209 to track.

Reply:
If we use `AltTypes`, does everything underneath just do the "right thing", or does it need additional changes?

Reply:
If you do similar to the `alarm_actions` example above, everything else should “just work”.